### DISCUSSION TITLE
API for adding repo to list of selected repos supported by Renovate

### DISCUSSION THREAD

**Discussion by srmadsen** [renovatebot/renovate#15455] [Request Help]
Does the Renovate App provide an API that would allow us to automate additions/deletions of repositories to the list of selected repositories being processed by Renovate App?

In our CICD system, the success of the Renovate PR build and unit tests are not sufficient to ensure success of the deployed application.  So, in our CICD system, we run acceptance tests against the deployed system in a DEV environment.  Frequently, the changes proposed by a renovate will pass the PR build, but fail after application is deployed. Renovate has no visibility into these failures, and Renovate may proceed to push other Renovate PRs into the broken deploy environment.

To prevent this, our CICD system checks to see if the commit that caused the deploy failure as a Renovate PR.  If se, we currently disable Renovate for that repository by adding an `"enabled": false` statement to the repo's renovate configuration to halt Renovate processing until the issue is resolved.  Once a successful deploy occurs, our CICD system will enable Renovate processing by removing that clause from the repo's Renovate configuration.

While the above works, we would prefer a solution that does not require changes to the repo's Renovate configuration files.  Ideally, the Renovate App would provide an API that allows to add/remove repos from the list of repos currently selected for Renovate processing.  Then we could add/delete repos from this list to enable/disable renovate processing without having to modify the repo's Renovate rules.

**Comment by rarkins** [comment]
You can use the GitHub API for this. Check out https://docs.github.com/en/rest/apps/installations#add-a-repository-to-an-app-installation for example. Using a PAT you can add or remove repositories from an installation.
